### DISCUSSION TITLE
Uploading Data to the Rubric for a Specific Assignment ...

### DISCUSSION THREAD

**Discussion by kspicer80** [ucfopen/canvasapi#537] [Q&A]
Hello Canvasapi folks,

I just wanted to say quickly what a time-saver this library has been for so far--even for a python neophyte like myself it's already saved me a great deal of work. I just had a quick question on a script that would be able to input scores on a rubric. Let's say I had a simple .json file with some data in it: a spot for the student_id, one for the score on the assignment, and then another for text comments to be uploaded to the submission. I know I can simply read that .json file into a dataframe and then write a simple loop to upload all of the data to the correct spots for each student in the course:

``` python
for index, row in df.iterrows():
    #assignment = course.get_assignment(row['assignment_id'])
    #user_submission = assignment.get_submission(row['student_id'])
    #user_submission.edit(submission=({'posted_grade': row['score']}))
    #user_submission.edit(comment={'text_comment': row['text_comments_to_upload']})
```

What if I wanted to use the rubric (already in the Canvas shell) to do the scoring? So, let's say I had two criteria in the rubric, "Criteria 1" and "Criteria 2." I can easily add some lines to the .json file that would keep the scores for each of the Criteria in the same way I'm keeping the information in the for loop above already.

I know there's a ```get_rubrics()``` already in the Canvasapi ... I know how to get the rubric ids as well from this function but I'm not sure about the method to use in the loop above to update the scores for each of the two criteria.

I hope this query makes some sense ...

Thanks a ton for all the hard work already on this fantastic library. It's streamlined my response to students' work in a way that I'm really grateful for, to be sure. 

Kevin

**Comment by bennettscience** [comment]
Kevin,
What you're asking to do is [possible via the API](https://canvas.instructure.com/doc/api/rubrics.html#method.rubric_assessments.create), but after a quick look through the library, that's an endpoint that hasn't been implemented yet.

In Canvas, it's called a `rubric_assessment` and can be passed in as JSON using the `Requester` class if you want to build your own function. The docs have a full explanation of the object to send, so once you build that, you can send a request with something like this (_not_ tested):

```python
from canvasapi import Canvas
from canvasapi.util import combine_kwargs

canvas = Canvas(URL, KEY)

# build your object as a dict
data = {}

request = canvas.__requester.request(
    'POST',
    'couses/{}/rubric_associations/{}/rubric_assessments'.format(course_id, rubric_association_id),
    _kwargs=combine_kwargs(data)
)
```

There's a template to create an issue to cover missing endpoints. Go ahead and submit an issue over there so it can get added to the backlog.
